### PR TITLE
Topic/first last isos

### PR DESCRIPTION
**Pull request by ethul**
Addition of isos for `First` and `Last`.

### CODE REVIEW

File: test/Main.purs [hunk 1]
```diff
@@ -3,9 +3,7 @@ module Test.Main where
 import Prelude
 
 import Data.Maybe
-import Data.Either
 import Data.Lens
-import Data.Tuple
 import Data.Traversable (Traversable)
 
 import Control.Monad.Eff.Console
```
Comment (garyb) [inline test/Main.purs:10]: Tests are failing as we're missing an `Eff` import here :smile:

**Comment by ethul** [comment]
Whoops! Thanks for catching this. The PR has been updated.

**Comment by garyb** [comment]
Thanks! Looks good to me :+1:

**Comment by ethul** [comment]
Welcome

**Comment by paf31** [comment]
This looks good, but let's consider some alternatives to growing a large collection of standard isos. Maybe something like the old `purescript-newtype` is the more general solution?

``` haskell
class Newtype n o where
  pack :: o -> n
  unpack :: n -> o

under :: forall n o. (Newtype n o) => (o -> n) -> IsoP n o
under _ = iso unpack pack

_First = under First
```

etc.?

**Comment by ethul** [comment]
The `Newtype` class approach looks great. Is `purescript-newtype` still around somewhere?

I am open to suggestions on the best way to go about these newtype isos.

**Comment by paf31** [comment]
No, it was unused so I got rid of it when cleaning up some old repos. This seems like a good reason to keep it though.

**Comment by ethul** [comment]
Shall I revive it?

If so, do you have a preference on the module name (e.g., `Data.Newtype`, `Data.Newtype.Class`, or something else)?

**Comment by paf31** [comment]
Sure :smile: 

Here is the entirety of the old code:

``` haskell
module Control.Newtype where

class Newtype n o where
  pack :: o -> n
  unpack :: n -> o

un :: forall n o. (Newtype n o) => (o -> n) -> n -> o
un _ = unpack
```

Perhaps `Data.Newtype` does make more sense. I stole the idea from the Haskell `newtype` package.

**Comment by ethul** [comment]
Thanks!

Also, I see there is [Control.Lens.Wrapped](https://hackage.haskell.org/package/lens-4.13/docs/Control-Lens-Wrapped.html). Do you think we should go with `Newtype` or something like `Wrapped` (i.e., should it be `Data.Lens.Wrapped` or `Data.Lens.Newtype`)?

**Comment by paf31** [comment]
Now that we have the `Newtype` class, we can handle this sort of thing generically!

**Comment by ethul** [comment]
Awesome!

**Pull request closed without merging**